src/OrleansServiceBus/Providers/Streams/EventHub/EventHubSequenceToken.cs

```diff
@@ -0,0 +1,47 @@
+﻿/*
+Project Orleans Cloud Service SDK ver. 1.0
+ 
+Copyright (c) Microsoft Corporation
+ 
+All rights reserved.
+ 
+MIT License
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and 
+associated documentation files (the ""Software""), to deal in the Software without restriction,
+including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+using System;
+using System.Globalization;
+using Orleans.Providers.Streams.Common;
+using Orleans.Streams;
+
+namespace Orleans.ServiceBus.Providers.Streams.EventHub
+{
+    [Serializable]
+    internal class EventHubSequenceToken : EventSequenceToken
+    {
+        public string Offset { get; private set; }
+
+        public EventHubSequenceToken(string offset, long sequenceNumber, int batchOffset)
```

Comment:
I will add comments.
Offset is required for finding this message in the event hub partition, but it can't be used for ordering or uniqueness (it's probably unique, it's just not provided for those purposes).  The sequence number is use for ordering/uniqueness of event hub messages, but each event hub message (batch) can have multiple events in it, so the batchOffset provides ordering and uniqueness for events within a batch.